quinn-proto/src/connection/mod.rs

```diff
@@ -773,6 +772,48 @@ impl Connection {
 
         self.app_limited = buf.is_empty() && !congestion_blocked;
 
+        // Send MTU probe if necessary
+        if buf.is_empty() && self.state.is_established() {
+            let space_id = SpaceId::Data;
+            let probe_size = match self
+                .path
+                .mtud
+                .poll_transmit(now, self.spaces[space_id].next_packet_number)
+            {
+                Some(next_probe_size) => next_probe_size,
+                None => return None,
```

Comment:
Not sure this `return None` makes sense, that would skip all the statistics stuff below? Might make sense to extract a method for the code below to avoid this drifting too far to the right.

Reply:
I think this is correct as written; we haven't sent anything, so there's no stats to update. Compare the `if buf.is_empty()` case that immediately follows the new code. Might still be opportunity for more clarity though.

Reply:
Ahh yes, I guess I forgot about the `buf.is_empty()` precondition here.